gen/aircraft.py

```diff
@@ -1158,17 +1165,34 @@ def build(self) -> MovingPoint:
         self.set_waypoint_tot(waypoint, self.timing.ingress)
         return waypoint
 
-class CasIngressBuilder(PydcsWaypointBuilder):
+class CasIngressBuilder(IngressBuilder):
     def build(self) -> MovingPoint:
         waypoint = super().build()
-        self.set_waypoint_tot(waypoint, self.timing.ingress)
-        waypoint.add_task(EngageTargets(max_distance=nm_to_meter(10),
+        ingress_waypoint = self.get_flight_point('INGRESS')
+        cas_waypoint = self.get_flight_point('CAS')
+        try:
+            ingress_point = Point(ingress_waypoint.position.x, ingress_waypoint.position.y)
+        
+            waypoint.add_task(EngageTargetsInZone(
+                                position=cas_waypoint,
+                                radius=ingress_point.distance_to_point(
+                                    cas_waypoint.position
+                                    ),
```

Comment:
40nm diameter seems huge to me, but that does appear to more or less match the width of our front lines. Better to use https://github.com/Khopa/dcs_liberation/blob/a465dde32fa9e6a49de2ec193277fb5b321f3f70/gen/conflictgen.py#L38 (divided by 2, of course) explicitly I think.

AI behaves okay with this? I'm assuming the nature of the task is that even though it's a huge area, since they're looking for targets they're not going to go wandering off into danger?

Reply:
Ah- I couldn't find where the frontline dimensions were set, that will be more robust.

The problem with the previous method (search and engage within 10nm) is that 10nm zone wanders with the unit, so if they go defensive due to fighters or SAMs or whatever, then they resume search wherever they end up (often pulling them further and further into danger).

Using the set zone, they return to where they belong.

Reply:
Fixed in fa01303